### PR TITLE
Schedule Editor: Consecutive Events

### PR DESCRIPTION
**Pull request by philrenaud**
- Adds the ability to set multiple events in a "Consecutive Booking" -- something like a panel interview, allowing for back-to-back meetings with different details.

- Removes the following properties from the top level of the schedule editor manifest and adds them to a new `events` array:
```
  event_title: string;
  event_description: string;
  event_location: string;
  event_conferencing: string;
  slot_size: 15 | 30 | 60;
  email_ids: string[];
```

- Introduces the concept of `host_rules` to dictate how multiple `email_ids` in a given event should be treated

image

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by pooja169usp — APPROVED** [review]

File: commons/src/types/ScheduleEditor.ts [hunk 1]
```diff
@@ -47,4 +41,20 @@ export interface Manifest extends NylasManifest {
   max_book_ahead_days: number;
   min_book_ahead_days: number;
   custom_fields: CustomField[];
+  events: any[]; // TODO
+}
+
+export interface EventDefinition {
```
Comment (ozsivanov) [inline commons/src/types/ScheduleEditor.ts:47]: Can we use the same interface as `Event` instead?

File: commons/src/types/ScheduleEditor.ts [hunk 2]
```diff
@@ -47,4 +41,20 @@ export interface Manifest extends NylasManifest {
...
+  event_description: string;
+  event_location: string;
+  event_conferencing: string;
+  slot_size: 15 | 30 | 60;
+  email_ids: string[];
+  host_rules: HostRules;
+}
+
+interface HostRules {
+  method: "all" | "user_determined" | "random";
```
Comment (ozsivanov) [inline commons/src/types/ScheduleEditor.ts:58]: What do each of these options do?

File: commons/src/types/ScheduleEditor.ts (hunk 1 above)
Comment (philrenaud) [reply commons/src/types/ScheduleEditor.ts:47]: As in the `_Event` type? My thinking was that we should keep a distinction between "Instructions for creating an event" and "event that exists and is temporaly situated"

**Review by ozsivanov — APPROVED** [review]

**Review by AaronDDM — APPROVED** [review]

File: commons/src/types/ScheduleEditor.ts (hunk 2 above)
Comment (philrenaud) [reply commons/src/types/ScheduleEditor.ts:58]: If you have multiple hosts for a meeting definition, for a given booked meeting:
- "all" will include all hosts
- "random" will randomly assign one of them
- ""user_determined" will let the booking user choose from among them

**Pull request merged**